libhypatia/src/segment.rs

```diff
@@ -0,0 +1,25 @@
+/// Call this macro from within each segment to include all the per-segment boilerplate.
+#[macro_export]
+macro_rules! define_segment {
+    () => {
+        mod _segment {
```

Comment:
It occurs to me that every instance of the `mod runtime;` statement in the base revision was conditioned by `#[cfg(not(test))]`; the `#[cfg_addr(not(test), no_mangle)]` on the definition of `pub extern "C" fn main()` is therefore superfluous, and so I think we can hide the entire module behind a `#[cfg(not(test))]`.

Also, given that the module doesn't pertain _just_ to segments but also to tasks, perhaps we could call it something else. I defaulted to `runtime` because I couldn't think of a better name, but given that this is all support stuff for no_std, perhaps we could just call the module, `_no_std` ?